File: push_sale_mobile-master/push_sale_mobile-master/lib/models/actor.dart
import 'package:push_sale/models/actor_profile.dart';
import 'package:push_sale/models/address.dart';
import 'package:push_sale/const/globals.dart' as global;
import 'package:push_sale/models/distributor.dart';

class Actor {
  final String id;
  final String firstname;
  final String lastname;
  final String mail;
  final String phone;
  final String image;
  final bool hasImage;
  final int rate;
  final ActorProfile? Profile;
  final Address? address;
  final Distributor? distributor;
  final Release? realisation;

  Actor({
    required this.id,
    required this.firstname,
    required this.lastname,
    required this.mail,
    required this.phone,
    required this.image,
    required this.hasImage,
    required this.rate,
    required this.Profile,
    required this.address,
    this.distributor,
    this.realisation,
  });

  static Actor fromMap(Map<String, dynamic> value) {
    return Actor(
        id: value["id"],
        firstname: value["firstname"],
        lastname: value["lastname"],
        mail: value["mail"],
        image:
            "${global.urlAPI}${value["image"] == "" || value["image"] == null ? "/storage/clients/no_image.jpg" : value["image"]}",
        hasImage: value["image"] != null && value["image"] != "",
        rate: int.parse(value["rate"].toString()),
        phone: value["phone"] ?? "",
        Profile: value["profile"] != null
            ? ActorProfile.fromMap(value["profile"])
            : null,
        address:
            value["address"] != null ? Address.fromMap(value["address"]) : null,
        distributor: value["distributor"] != null
            ? Distributor.fromMap(value["distributor"])
            : null,
        realisation: value["realisation"] != null
            ? Release.fromMap(value["realisation"])
            : null);
  }
}

class Release {
  final int orders_count;
  final double orders_amount;
  Release({
    required this.orders_count,
    required this.orders_amount,
  });

  static Release fromMap(Map<String, dynamic> value) {
    print(value);
    return Release(
      orders_count: int.parse(value["orders_count"].toString()),
      orders_amount: double.parse(value["orders_amount"].toString()),
    );
  }
}

File: push_sale-master/app/Models/Actor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Passport\HasApiTokens;

class Actor extends Model
{
    use HasApiTokens, HasFactory;
    protected $table = "actor";

    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        "id",
        "type",
        "firstname",
        "user_id",
        "lastname",
        "mail",
        "profile_id",
        "rate",
        "phone",
        "image",
        "distributor_id",
        "address_id",
        "is_active"
    ];

    protected $hidden = ["user_id", "address_id", "profile_id", "created_at", "updated_at"];

    public function Address()
    {
        return $this->belongsTo(Address::class);
    }

    public function Profile()
    {
        return $this->hasOne(ActorProfile::class, "id", "profile_id");
    }

    public function Clients()
    {
        return $this->hasMany(Client::class, "actor_id", "id");
    }

    public function User()
    {
        return $this->hasOne(User::class, "id", "user_id");
    }

    public function StockMobile()
    {
        return $this->hasOne(StockMobile::class, "actor_id", "id");
    }
    public function Distributor()
    {
        return $this->belongsTo(Distributor::class);
    }

    public function realisation(){
        return $this->hasMany(Order::class);
    }
}
